notebooks/MixedModelsTutorial_basics.jl

```diff
@@ -66,124 +71,207 @@ Number of scores: 525126
+	recode!(df.Test, "Run" => "Endurance", "Star_r" => "Coordination",
+		               "S20_r" => "Speed", "SLJ" => "PowerLOW", "BPT" => "PowerUP")
 	describe(df)
 end
 
+# ╔═╡ 51b77ca3-374d-4701-b936-b4a40834cf70
+md"""
+#### 1.2.2 Transformations
+We center `age` at 8.5 years and compute z-scores for each `Test`. With these variables the data frame `df` contains all variables used for the final model in the original publication.
+"""
+
+# ╔═╡ 7440c439-9d4c-494f-9ac6-18c9fb2fe144
+begin
+	transform!(df, :age, :age => (x -> x .- 8.5) => :a1); # centered age (linear)
+	select!(groupby(df,  :Test), :, :score => zscore => :zScore); # z-score
+	dfx = df[:, Not(:score)]   # drop score
+end
+
 # ╔═╡ 1e6fe7f6-9ce4-4b13-953b-bf7dc532e53a
 md"""
 #### 1.2.3 Extract a stratified subsample 
 
-We extract a random sample of 1000 children from the Sex (2)  x Test (5) cells of the design. Cohort and School are random. 
+For the prupose of the tutorial, we extract a random sample of 1000 boys and 1000 girls. `Child`, `School`, and `Cohort` are grouping variables. Traditionally, they are called random factors because the units (levels) of the factor are assumed to be a random sample from the population of their units (levels).  
+
+Cohort has only nine "groups" and could have been included as a set of polynomical fixed-effect contrasts rather than a random factor. This choice warrants a short excursion: The secular trends are very different for different tests and require the inclusion of interaction terms with `Test` contrasts (see Figure 4 in [Fühner et al. (2021)](https://rdcu.be/cwSeR). The authors opted to absorb these effects in cohort-related variance components for the `Test` contrasts and plan to address the details of secular changes in a separate analysis.
+
+For complex designs, when they are in the theoretical focus of an article, factors and covariates should be specified as part of the fixed effects. If they are not in the theoretical focus, but serve as statistical control variables, they could be put in the RES - if supported by the data.
+"""
+
+# ╔═╡ 92038a7c-69f6-445d-ba29-03ae4cf9f115
+md"**Unstack:** Data are in the long format. We need them in a child-based wide format -- one row per child"
+
+# ╔═╡ fe027497-7ed1-4780-871b-547d4f466072
+df_w = unstack(dfx, :Test, :zScore)
+
+# ╔═╡ 29d0a540-29ce-43de-9dfe-75e535ff94eb
+md""" **Stratified sampling:** `MersenneTwister(42)` specifies `42` as the seed for the random number generator to ensure reproducibility of the stratification. The stratified sample is stored in data frame `dat`.
 """
 
-# ╔═╡ 6e3adecf-98ff-4437-8e16-3854381ca38b
+# ╔═╡ b3f32059-0d7b-4b66-8559-7c8a9f4d14d3
+
 begin
-    dat =
-	@chain df begin
-  	  @transform(:Sex = :Sex == "female" ? "Girls" : "Boys")
-   	  @groupby(:Test, :Sex)
-   	  combine(x -> x[sample(1:nrow(x), 1000), :])
+dat_w =
+	@chain df_w begin
+   	  @groupby(:Sex)
+   	  combine(x -> x[sample(MersenneTwister(42), 1:nrow(x), 1000), :])
```

Comment:
I believe this will create a new random number generator at the same seed for Girls and for Boys.  A more common approach is to initialize the random number generator outside the `groupby` loop.
Also, the `@chain` macro call is a single statement and doesn't need to be enclosed in a `begin/end` block.
I would probably write it as
```jl
dat = let rng = MersenneTwister(42)
    @chain df begin
        groupby(:Sex)
        combine(x -> x[rand(rng, 1:nrow(x), 1000), :])
    end
end
```
or even
```jl
dat = let rng = MersenneTwister(42)
    combine(groupby(df, :Sex), x -> x[rand(rng, 1:nrow(x), 1000), :])
end
```

Reply:
I adopted the second proposal.

Reply:
I adopted the next proposal.